src/MakoIoT.Device/IoTDevice.cs

```diff
@@ -14,22 +17,30 @@ public class IoTDevice : IDevice
         public IoTDevice(IServiceProvider serviceProvider)
         {
             ServiceProvider = serviceProvider;
+
+            if (IsRegistered(ServiceProvider, typeof(ILoggerFactory)))
```

Comment:
Why can't we just inject logger into constructor?

Reply:
My thinking here was for two reasons:

1. `IoTDevice` doesn't currently require that an `ILogger` be registered so injecting it directly might be a breaking change for some users (lots of other components downstream do directly require an `ILogger` though so this might not be a large concern).
2. I prefer to use `ILoggerFactory` as it gives me the most flexibility. In debug mode I can get a unique logger per class and in release mode I can re-use a shared logger if I want.

Reply:
With the upcoming changes in logger I would rather to stick with "ILogger" in constructor (I'm going to introduce Serilog approach with multiple Sinks, goal for release is to support Console and SdCard) . Also that's how we are using logger in other classes.

Reply:
Sounds good.

I updated IoTDevice.ctor and the associated unit tests.